server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2128,8 +2128,7 @@ public BlobStoreIndexShardSnapshot loadShardSnapshot(BlobContainer shardContaine
             return new Tuple<>(shardSnapshots, latest);
         } else if (blobs.stream().anyMatch(b -> b.startsWith(SNAPSHOT_PREFIX) || b.startsWith(INDEX_FILE_PREFIX)
                                                                               || b.startsWith(UPLOADED_DATA_BLOB_PREFIX))) {
-            throw new IllegalStateException(
-                "Could not find a readable index-N file in a non-empty shard snapshot directory [" + shardContainer.path() + "]");
+            logger.warn("Could not find a readable index-N file in a non-empty shard snapshot directory [" + shardContainer.path() + "]");
```

Comment:
I thought about whether this is ok or not quite a bit. I think we're not introducing any new risk qualitatively really. If we can't read what is in this directory it's broken, period.

For snapshot creation:
We can have some freak concurrency issues here if we aren't yet using shard generations and two data nodes try to write to the same path at the same time. But those we resolve either (on S3) by using the 3 minute wait or by not allowing overwrites for index-N on other types of repos. With shard generations being used, this is a non-issue anyway.
For deletions we are in the clear also, because everything runs on the master.